pkg/controller/controllercmd/builder.go

```diff
@@ -157,6 +169,14 @@ func (b *ControllerBuilder) WithLeaderElection(leaderElection configv1.LeaderEle
 	return b
 }
 
+// WithLeaderElectionResourceLock sets the resource lock. If not set, controllercmd.ConfigMapsLeasesResourceLock will
+// be used for backwards compatibility.
+func (b *ControllerBuilder) WithLeaderElectionResourceLock(resourceLock leaderElectionResourceLock) *ControllerBuilder {
```

Comment:
instead of using the lock as a parameter, can we have two methods?
```
WithLeaderElectionLeaseResourceLock() *ControllerBuilder
WithLeaderElectionConfigmapLeaseResourceLock() *ControllerBuilder
```

this way we don't have to have validation around resource lock names and in future we can simply remove `WithLeaderElectionConfigmapLeaseResourceLock`